apps/examples/rt_ocf_test/rt_ocf_test_main.c

```diff
@@ -0,0 +1,738 @@
+	g_rt_isWiFiConnected = 1;
+	printf("\n[WiFi] Connected!\n");
+	pthread_cond_signal(&g_rt_wifi_condition);
+	pthread_mutex_unlock(&g_rt_wifi_mutex);
+}
+
+static void wm_sta_disconnected(void)
+{
+	g_rt_isWiFiConnected = 0;
+	printf("\n[WiFi] Disconnected!\n");
+}
+
+void wm_softap_sta_join(void)
+{
+}
+
+void wm_softap_sta_leave(void)
+{
+}
+
+void wm_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res)
+{
+}
+
+#define TAG "SIMPLE_SERVER"
+
+typedef struct {
+	const char *uri;
+	ocf_resource_s resource;
+	bool value;
+	ocf_request_cb get;
+	ocf_request_cb put;
+	ocf_request_cb post;
+	ocf_request_cb delete;
+} ocf_item;
+
+ocf_item blue_led = {.uri = "/light/blue" };
+ocf_item red_led = {.uri = "/light/red" };
+
+static ocf_rep_encoder_s create_item_representation(ocf_item *item)
+{
+	ocf_rep_encoder_s rep = ocf_rep_encoder_init(OCF_REP_MAP);
+	if (!rep) {
+		printf("[%s]Can not make rep!! \n", __func__);
+		return NULL;
+	}
+
+	ocf_rep_add_bool_to_map(rep, "value", item->value);
+	return rep;
+}
+
+static void notifiy_item_state(ocf_item *item)
+{
+	ocf_rep_encoder_s rep = create_item_representation(item);
+	ocf_observe_notify(item->uri, rep);
+	ocf_rep_encoder_release(rep);
+	printf("[%s]Item Value Changed!!!!\n", __func__);
+}
+
+static ocf_result_t apply_interface_to_representation(ocf_request_s request, ocf_resource_s resource, ocf_rep_encoder_s rep)
+{
+	if (rep) {
+		// ocf_request_query_set_s *queries = ocf_request_get_queries(request);
+		ocf_res_add_if_rt_rep(rep, resource);
+		return OCF_OK;
+	}
+	return OCF_ERROR;
+}
+
+static void blue_led_get_handler_func(ocf_request_s request, ocf_rep_decoder_s data)
+{
+	printf("Blue LED Receive GET Method\n");
+	ocf_item *item = &blue_led;
+
+	ocf_rep_encoder_s rep = create_item_representation(item);
+	ocf_result_t ret = apply_interface_to_representation(request, item->resource, rep);
+
+	if (ret == OCF_OK) {
+		ocf_response_send(request, rep, OCF_RESPONSE_CONTENT);
+		ocf_rep_encoder_release(rep);
+	} else {
+		ocf_response_send(request, NULL, OCF_RESPONSE_RESOURCE_NOT_FOUND);
+	}
+}
+
+static void red_led_get_handler_func(ocf_request_s request, ocf_rep_decoder_s data)
+{
+	printf("Red LED Receive GET Method\n");
+	ocf_item *item = &red_led;
+
+	ocf_rep_encoder_s rep = create_item_representation(item);
+	ocf_result_t ret = apply_interface_to_representation(request, item->resource, rep);
+	if (ret == OCF_OK) {
+		ocf_response_send(request, rep, OCF_RESPONSE_CONTENT);
+		ocf_rep_encoder_release(rep);
+	} else {
+		ocf_response_send(request, NULL, OCF_RESPONSE_RESOURCE_NOT_FOUND);
+	}
+}
+
+static void put_handler_func(ocf_request_s request, ocf_rep_decoder_s data)
+{
+	printf("Put request is received\n");
+	ocf_response_send(request, NULL, OCF_RESPONSE_METHOD_NOT_ALLOWED);
+	return;
+}
+
+static int gpio_read(int port)
+{
+	char buf[4];
+	char devpath[16];
+	snprintf(devpath, 16, "/dev/gpio%d", port);
+	int fd = open(devpath, O_RDWR);
```

Comment:
return value of open & ioctl & read & write should be checked to prevent error

Reply:
Error handling code is added for each function call.